src/core/Basics/Song.h

```diff
@@ -60,12 +60,16 @@ class Song : public H2Core::Object
 			SONG_MODE
 		};
 
+		static constexpr int nDefaultResolutionTPQN = 48;
```

Comment:
Only because that's what MIDI uses, and that's what the core already uses (because it looks like it was heavily influenced by MIDI). Multiplying up by 4 when needed (seems to be a fairly small number of uses) is a fairly small price to pay for the consistency (if not, a macro can be defined).

Reply:
Ok.
a macro would be good enough